payplug/__init__.py

```diff
@@ -41,8 +42,10 @@ def set_api_version(version):
     :type version: string
     """
 
-    if not isinstance(version, string_types):
-        raise exceptions.ConfigurationError("Expected string value for version.")
+    try:
+        date = datetime.strptime(version, '%Y-%m-%d')
+    except ValueError:
```

Comment:
L'appel à `strptime` va raise une `ValueError` si tu passes une string invalide, mais une `TypeError` si tu passes n'importe quoi (None, 42, etc.)
Du coup idéalement il faudrait aussi attraper les `TypeError` et mettre dans le message "Version should be a string formatted as etc."

Par ailleurs c'est inutile de créer une variable `date` si tu ne l'utilises pas derrière.

Après j'arrête de pinailler promis :)